### PR TITLE
Set curswant in nvim_win_set_cursor

### LINKED ISSUE #8591: After setting cursor position through python API, up/down does not keep the same column
- `nvim --version`:
```
NVIM v0.2.2
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -Wconversion -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/builddir/build/BUILD/neovim-0.2.2/build/config -I/builddir/build/BUILD/neovim-0.2.2/src -I/builddir/build/BUILD/neovim-0.2.2/.deps/usr/include -I/builddir/build/BUILD/neovim-0.2.2/.deps/usr/include -I/builddir/build/BUILD/neovim-0.2.2/.deps/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/builddir/build/BUILD/neovim-0.2.2/build/src/nvim/auto -I/builddir/build/BUILD/neovim-0.2.2/build/include
Compiled by mockbuild

Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
- Vim (version: ) behaves differently?
No
- Operating system/version:
Centos 7.5.1804 
- Terminal name/version:
tmux 1.8
- `$TERM`:
screen-256color

### Steps to reproduce using `nvim -u NORC`

Save this as test.vim:
```
function! s:Test()
python3 << EOF
import vim
vim.current.window.cursor = (12, 15)
EOF
endfunc

command! Test call s:Test()

"Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
"tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam,
"quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
"consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
"cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non
"proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
```
Then

* `nvim -u NORC`
* `:edit test.vim`
* `:source test.vim`
* navigate the cursor to any column other than column 16
* :Test
* cursor is now correctly in line 12 column 16
* `j`
* cursor is now back in whatever column it was in before

### Actual behaviour

Cursor reverts to whatever column it came from.

### Expected behaviour

Cursor stays in the column that it was moved to.

### PR DESCRIPTION
**Pull request by dimbleby**
Fixes #8591

### CODE REVIEW

**Comment by justinmk** [comment]
@dimbleby Thank you very much for looking into this, it's a big help.

Are you able to write a test? See [here](https://github.com/neovim/neovim/blob/master/test/README.md#filter-by-file), it's basically just a matter of:

1. Add a new `it()` case to the `describe('{get,set}_cursor',` block of `test/functional/api/window_spec.lua`.
2. Run the test like this (see [here](https://github.com/neovim/neovim/blob/master/test/README.md#filter-by-file) for other options):
    ```
    TEST_FILE=test/functional/api/window_spec.lua make functionaltest
    ```

File: src/nvim/api/window.c [hunk 1]
```diff
@@ -93,6 +93,9 @@ void nvim_win_set_cursor(Window window, ArrayOf(Integer, 2) pos, Error *err)
   // When column is out of range silently correct it.
   check_cursor_col_win(win);
 
+  // Make sure we stick in this column.
+  win->w_curswant = win->w_cursor.col;
```
Comment (bfredl) [inline src/nvim/api/window.c]: why not use  `col` here directly? If we set to column 20 a line with 10 columns, and then `j` to a line with 20 columns, is it not more reasonable to then use column 20?

**Comment by dimbleby** [comment]
@justinmk thanks for the pointer, I added a testcase

File: src/nvim/api/window.c (hunk 1 above)
Comment (dimbleby) [reply src/nvim/api/window.c]: Sure, I'm fine with that.  (Actually that was my first draft, and then I changed my mind - but for no very good reason).  I'll change it.

**Pull request merged**